### PR TITLE
chore: address clippy lints and bump ci clippy stable

### PR DESCRIPTION
**Pull request by jxs**
## Description
superseeds #5248, sorry @drHuangMHT only saw yours now, can you review this one?
https://github.com/libp2p/rust-libp2p/actions/runs/8377129630/job/22938498078?pr=5253#step:6:843 is a bug, see [here](https://github.com/rust-lang/rust-clippy/issues/12377)
Thanks!

### CODE REVIEW

**Comment by drHuangMHT** [comment]
I'm not sure if the `mixed_attribute_style` lint works properly because `include!` essentially copies entire content of the source file. Though it probably doesn't matter because it's a generated file.
Also I'm not super familiar with the codebase so I can't really comment on some changes to the unused code.

**Comment by drHuangMHT** [comment]
>https://github.com/libp2p/rust-libp2p/actions/runs/8377129630/job/22938498078?pr=5253#step:6:843 is a bug, see [here](https://github.com/rust-lang/rust-clippy/issues/12377)

Someone's going to file a backport on beta like what they did with `unconditional_recurison`. So It's not something they'd do very soon.

File: swarm/src/connection.rs [hunk 1]
```diff
@@ -734,7 +731,7 @@ enum Shutdown {
     /// A shut down is planned as soon as possible.
     Asap,
     /// A shut down is planned for when a `Delay` has elapsed.
-    Later(Delay, Instant),
```
Comment (jxs) [inline swarm/src/connection.rs:737]: clippy reported `Instant` as not being read anywhere

File: swarm/src/connection.rs [hunk 2]
```diff
@@ -1000,7 +991,9 @@ mod tests {
             self: Pin<&mut Self>,
             _: &mut Context<'_>,
         ) -> Poll<Result<Self::Substream, Self::Error>> {
-            Poll::Ready(Ok(PendingSubstream(Arc::downgrade(&self.counter))))
+            Poll::Ready(Ok(PendingSubstream {
+                _weak: Arc::downgrade(&self.counter),
```
Comment (jxs) [inline swarm/src/connection.rs:995]: clippy reported `PendingSubstream`s `Weak<()>` was not being read and suggested replacing to `PendingSubstream(())` but removing the weak reference fails tests, I used this pattern instead.

**Review by thomaseizinger — APPROVED** [review]
LGTM

Nice cleanup :)

**Comment by jxs** [comment]
thanks Thomas!

**Pull request merged**